src/Pretzel.Logic/Commands/BaseParameters.cs

```diff
@@ -21,20 +24,27 @@ public sealed class BaseParameters
         [Export("SourcePath")]
         public string Path { get; private set; }
 
-        [Export]
         public IFileSystem FileSystem { get; private set; }
 
         public List<string> CommandArgs { get; private set; }
 
         private BaseParameters(string[] arguments, IFileSystem fileSystem)
+        {
+            SetDefaults(arguments, fileSystem);
+        }
+
+        [ImportingConstructor]
+        public BaseParameters(IFileSystem fileSystem) : this(Environment.GetCommandLineArgs().Skip(1).ToArray(), fileSystem) { }
+
+        private void SetDefaults(string[] arguments, IFileSystem fileSystem)
         {
             Options = new OptionSet
                 {
-                    {"help", "Display help mode", p => Help = true},
-                    {"debug", "Enable debugging", p => Debug = true},
+                    { "help", "Display help mode", p => Help = true },
+                    { "debug", "Enable debugging", p => Debug = true },
                     { "safe", "Disable custom plugins", v => Safe = true },
                     { "d|directory=", "[Obsolete, use --source instead] The path to site directory", p => Path = p },
-                    { "s|source=", "The path to the source site (default current directory)", p => Path = p}
+                    { "s|source=", "The path to the source site (default current directory)", p => Path = Path = p }
```

Comment:
There is a redundancy of Path affectation.

Reply:
Good catch again, fixed.